src/reference.rs

```diff
@@ -133,6 +157,15 @@ impl Reference {
         self.digest.as_deref()
     }
 
+    /// ns returns the original registry when pulled via a mirror.
+    pub fn ns(&self) -> Option<&str> {
```

Comment:
Can this just be called `namespace` to match the proposal? I know it maps to the parameter, but it is a bit confusing when reading